packages/common/src/hooks/useImageSize2.ts

```diff
@@ -0,0 +1,135 @@
+import { useState, useEffect, useCallback } from 'react'
```

Comment:
add some unit tests

Reply:
I spent a while here trying to figure out how to get some tests and kept running into horrible roadblocks between react testing library, jsdom, vitest, and misc. mocking. In order to not block this PR, which feels like such a win, I'd like to merge first and revisit that down the line